llvm/test/Transforms/LoopRotate/dbgvalue.ll

```diff
@@ -38,12 +61,81 @@ return:                                           ; preds = %if.end
   ret i32 %z.tr, !dbg !17
 }
 
-define i32 @tak2(i32 %x, i32 %y, i32 %z) nounwind ssp !dbg !21 {
-; CHECK-LABEL: define i32 @tak2(
+; Repeat of the tak function, with only one DILocalVariable, checking that we
+; don't insert duplicate debug intrinsics. The initial duplicates are preserved.
+; FIXME: this test checks for the de-duplication behaviour that loop-rotate
+; has today, however it might not be correct -- should not the _last_
+; assignment to the variable be preserved, not the first?
```

Comment:
Which dbg.value are you describing in the FIXME?  Looking at the CHECK lines above (for the `tak` function) it looks like the last dbg.value _is_ preserved?

Reply:
Comment updated to specify it more directly.

Reply:
I see, it's the reverse-insertion + strange deduplication combo.